src/roomsync.rs

```diff
@@ -600,6 +605,7 @@ impl YRoom {
                     }
                 }
                 Message::Sync(SyncMessage::Update(data)) => {
+                    has_edits = true;
```

Comment:
`has_edits = true` should also be added on the other `txn.apply_update(update)` in the sync step 2, right?

Reply:
Added to the sync step 2 update as well. Is it correct to enable the flag before the update match? Or should this happen inside the `Ok(update) {...}` scope?

Reply:
Right. So even if there is an error, the document *has changes* albeit in a faulty way?

Reply:
No, I mean if the update cannot be decoded correctly, it will not be applied to the document, therefore no changes.